src/ring_theory/perfection.lean

```diff
@@ -126,13 +140,18 @@ instance perfect_ring : perfect_ring (ring.perfection R p) p :=
   frobenius_pth_root' := congr_fun $ congr_arg ring_hom.to_fun $ @frobenius_pth_root R _ p _ _,
   pth_root_frobenius' := congr_fun $ congr_arg ring_hom.to_fun $ @pth_root_frobenius R _ p _ _ }
 
+/-- The perfection of a ring `R` with characteristic `p`, as a subring,
+defined to be the projective limit of `R` using the Frobenius maps `R → R`
+indexed by the natural numbers, implemented as `{ f : ℕ → R | ∀ n, f (n + 1) ^ p = f n }`. -/
+def _root_.ring.perfection_subring (R : Type u₁) [comm_ring R] [char_p R p] : subring (ℕ → R) :=
```

Comment:
Rather than needing root, would it make sense to move this to the top of the file?

Reply:
Done